src/box/txn_limbo.c

```diff
@@ -597,6 +600,7 @@ txn_limbo_read_promote(struct txn_limbo *limbo, uint32_t replica_id,
 	limbo->owner_id = replica_id;
 	limbo->confirmed_lsn = vclock_get(&limbo->confirmed_vclock,
 					  replica_id);
+	limbo->entry_to_confirm = NULL;
```

Comment:
That looks a bit dangerous. Intuitively I would expect that after `txn_limbo_read_confirm()` the pointer is already updated. But here you update it afterwards manually, inside `txn_limbo_read_promote()`.

Currently it works, but if replicas would start collecting acks, they would have `entry_to_confirm` non-NULL and not updated when the replicas receive a CONFIRM.

Anyway, not principal to me if we fix it now.

Reply:
Let's not fix this now.